daml-lf/transaction-test-lib/src/main/scala/lf/transaction/test/TransactionBuilder.scala

```diff
@@ -81,81 +74,62 @@ final class TransactionBuilder(
 
   def buildCommitted(): CommittedTransaction = CommittedTransaction(build())
 
-  def newCid: ContractId = ContractId.V1(newHash())
+  def newCid: ContractId = TransactionBuilder.newV1Cid
 
-  def versionContract(contract: ContractInst[Value]): ContractInst[TxValue] =
-    ContractInst.map1[Value, TxValue](transactionValue(contract.template))(contract)
+  def versionContract(contract: Value.ContractInst[Value]): value.Value.ContractInst[TxValue] =
+    Value.ContractInst.map1[Value, TxValue](transactionValue(contract.template))(contract)
 
   private[this] def transactionValue(templateId: Ref.TypeConName): Value => TxValue =
     value.Value.VersionedValue(pkgTxVersion(templateId.packageId), _)
 
-  def create(
-      id: String,
-      template: String,
-      argument: Value,
-      signatories: Seq[String],
-      observers: Seq[String],
-      key: Option[Value],
-  ): Create =
-    create(ContractId.assertFromString(id), template, argument, signatories, observers, key)
-
   def create(
       id: ContractId,
-      template: String,
+      templateId: Ref.Identifier,
       argument: Value,
-      signatories: Seq[String],
-      observers: Seq[String],
-      key: Option[Value],
+      signatories: Set[Ref.Party],
+      observers: Set[Ref.Party],
+      key: Option[Value] = None,
```

Comment:
Hi @remyhaemmerle-da , these looks like good changes and simple simplifications to make. Just wanted to make you aware that these will require me to adapt these canton-side tests (that I am still working on) - just as a heads-up that we are beginning to rely on `TransactionBuilder` more and more: https://github.com/DACH-NY/canton/pull/6969